signature/json.go

```diff
@@ -122,3 +64,25 @@ func paranoidUnmarshalJSONObject(data []byte, fieldResolver func(string) interfa
 	}
 	return nil
 }
+
+// paranoidUnmarshalJSONObject unmarshals data as a JSON object, but failing on the slightest unexpected aspect
+// (including duplicated keys, unrecognized keys, and non-matching types). Each of the fields in exactFields
+// must be present exactly once, and none other fields are accepted.
+func paranoidUnmarshalJSONObjectExactFields(data []byte, exactFields map[string]interface{}) error {
+	seenKeys := map[string]struct{}{}
```

Comment:
I somewhat prefer the `struct{}` to make the type closer to the true type (set of `string`), see also earlier conversation in  https://github.com/containers/image/pull/223 , and perhaps earlier.

I don’t feel all that strongly about this, but if you can live with it, I’ll merge as is.